haskell/experimental/private/module.bzl

```diff
@@ -63,6 +69,15 @@ def haskell_module_impl(ctx):
         # to debug issues in non-sandboxed builds.
         "-Wmissing-home-modules",
     ])
+    if with_profiling:
+        args.add_all([
+            "-prof",
+            "-fexternal-interpreter",
+            "-hisuf",
+            "p_hi",
+            "-osuf",
+            "p_o",
```

Comment:
Are those still required when `-o` and `-ohi` are being set explicitly?

Reply:
Yes! This was exactly the headache I caused myself, because I thought the same thing, removed this, and the more complicated tests started failing.

It will generate the correct files even without this, but it won't *look* for the right files when you pass in other profiling-enabled modules as dependencies. It will look for `.hi` files instead of `.p_hi` files.

Reply:
Also worth noting that `haskell_library()` does both set `-o`, `-ohi`, and `-osuf`, `-hisuf`

Reply:
> It will generate the correct files even without this, but it won't _look_ for the right files when you pass in other profiling-enabled modules as dependencies. It will look for `.hi` files instead of `.p_hi` files.

That makes sense, thanks for clarifying!